### PR TITLE
`/tokens` page

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Tokens page implementation #45 
#### Which issue(s) does this PR fixes?:

Fixes #

## Screenshot

image

### CODE REVIEW

**Comment by siradji** [comment]
@fuxingloh

**Comment by fuxingloh** [comment]
add e2e tests

File: src/layouts/components/Header.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ export function Header (): JSX.Element {
               <div className='hidden md:flex flex-wrap'>
                 <HeaderLink className='ml-12' text='Dex' pathname='/dex' />
                 <HeaderLink className='ml-4' text='Prices' pathname='/prices' />
+                <HeaderLink className='ml-4' text='Tokens' pathname='/Tokens' />
```
Comment (fuxingloh) [inline src/layouts/components/Header.tsx:51]:
```suggestion
                <HeaderLink className='ml-4' text='Tokens' pathname='/tokens' />
```

File: src/pages/tokens/index.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      setItems([...(items.sort((a, b) => parseInt(a.minted) - parseInt(b.minted)))])
+    }
+  }
+
+  return (
+    <div className='container mx-auto px-4 pt-12 pb-20'>
+      <Head
+        title='Tokens'
+      />
+      <>
```
Comment (fuxingloh) [inline src/pages/tokens/index.tsx:40]: remove unnecessary `<>`, creating unnecessary indents

File: src/pages/tokens/index.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        <div className='flex justify-between items-center'>
+          <h1 className='text-2xl font-semibold'>Tokens</h1>
+          <div className='space-x-2'>
+            <button className='p-3 rounded bg-gray-50 border border-grey-200 shadow-sm'>
+              <FaSearch height={40} />
+            </button>
+            <button className='p-3 rounded bg-gray-50 border border-grey-200 shadow-sm'>
+              <CgSortAz height={40} />
+            </button>
```
Comment (fuxingloh) [inline src/pages/tokens/index.tsx:49]: remove, not planning to support these anytime soon

File: src/pages/tokens/index.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          </div>
+        </div>
+        <AdaptiveTable className='mt-6'>
+          <AdaptiveTable.Header>
+            <AdaptiveTable.Head className='flex items-center'>
+              TOKEN NAME <button data-sort-by='tokenName' onClick={(e) => handleSort(e)}><FaSort /></button>
```
Comment (fuxingloh) [inline src/pages/tokens/index.tsx:55]: remove sort, not planning to support these, we are not indexing theses columns

File: src/pages/tokens/index.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+function TokenRow ({ data }: { data: TokenData }): JSX.Element {
+  const TokenIcon = getTokenIcon(data.symbol)
+  return (
+    <AdaptiveTable.Row>
+      <AdaptiveTable.Cell title='TOKEN NAME' className='align-middle'>
+        <div className='flex items-center space-x-2'>
+          <TokenIcon />
+          <Link href={{ pathname: `/tokens/${data.id}` }}>
```
Comment (fuxingloh) [inline src/pages/tokens/index.tsx:79]: remove feature that are not part of `/tokens`

File: src/pages/tokens/index.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  const [items, setItems] = useState(tokens.items)
+
+  function handleSort (e: MouseEvent<HTMLButtonElement>): void {
+    const sortBy = e.currentTarget.dataset.sortBy
+    if (sortBy === 'tokenName') {
+      setItems([...(items.sort((a, b) => a.name.localeCompare(b.name)))])
+    } else {
+      setItems([...(items.sort((a, b) => parseInt(a.minted) - parseInt(b.minted)))])
+    }
+  }
```
Comment (fuxingloh) [inline src/pages/tokens/index.tsx:33]: remove

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: src/pages/tokens/index.tsx (hunk 4 above)
Comment (siradji) [reply src/pages/tokens/index.tsx]: Understood

File: src/pages/tokens/index.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export default function TokensPage ({ tokens }: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const [items] = useState(tokens.items)
+
+  return (
+    <div className='container mx-auto px-4 pt-12 pb-20'>
+      <Head
+        title='Tokens'
+      />
```
Comment (canonbrother) [inline src/pages/tokens/index.tsx:26]: hmm
```suggestion
      <Head title='Tokens'/>
```

File: src/pages/tokens/index.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export async function getServerSideProps (context: GetServerSidePropsContext): Promise<GetServerSidePropsResult<TokensPageData>> {
+  const items = await getWhaleApiClient(context).tokens.list()
+  return {
+    props: {
+      tokens: {
+        items,
+        nextToken: items.nextToken ?? null
+      }
```
Comment (canonbrother) [inline src/pages/tokens/index.tsx:73]: i think missing `hasNext`, can you confirm??

Comment (siradji) [reply src/pages/tokens/index.tsx:73]: nextToken alone will be sufficient.

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**